category/category.go

```diff
@@ -0,0 +1,269 @@
+	LoadAllRelationshipsOfCategory(ctx context.Context, categoryID uuid.UUID) ([]*WorkItemTypeCategoryRelationship, error)
+	Save(ctx context.Context, category *Category) (*Category, error)
+}
+
+// NewRepository creates a new storage type.
+func NewRepository(db *gorm.DB) Repository {
+	return &GormRepository{db: db}
+}
+
+// GormRepository is the implementation of the storage interface for Categories.
+type GormRepository struct {
+	db *gorm.DB
+}
+
+// List all Categories.
+func (m *GormRepository) List(ctx context.Context) ([]*Category, error) {
+	var objs []*Category
+	err := m.db.Find(&objs).Error
+	if err != nil && err != gorm.ErrRecordNotFound {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "unable to list categories")
+		return nil, errs.WithStack(err)
+	}
+	return objs, nil
+}
+
+// CreateRelationship creates relationship between workitemtype and category
+func (m *GormRepository) CreateRelationship(ctx context.Context, relationship *WorkItemTypeCategoryRelationship) error {
+	db := m.db.Create(relationship)
+	if db.Error != nil {
+		if gormsupport.IsUniqueViolation(db.Error, "work_item_type_categories_idx") {
+			return errors.NewBadParameterError("category+workitemtype", relationship.CategoryID).Expected("unique")
+		}
+		log.Error(ctx, map[string]interface{}{
+			"category_id": relationship.CategoryID,
+			"wit_id":      relationship.WorkItemTypeID,
+			"err":         db.Error.Error(),
+		}, "unable to create workitemtype category relationship")
+		return errors.NewInternalError(db.Error)
+	}
+	return nil
+}
+
+// Create creates category. This function is used to populate categories table during migration -> PopulateCategories()
+func (m *GormRepository) Create(ctx context.Context, category *Category) (*Category, error) {
+	if strings.TrimSpace(category.Name) == "" {
+		return nil, errors.NewBadParameterError("Name", category.Name)
+
+	}
+	db := m.db.Create(category)
+	if db.Error != nil {
+		if gormsupport.IsUniqueViolation(db.Error, "categories_name_idx") {
+			return nil, errors.NewBadParameterError("Name", category.Name).Expected("unique")
+		}
+		log.Error(ctx, map[string]interface{}{
+			"category_id": category.ID,
+			"err":         db.Error.Error(),
+		}, "unable to create category")
+		return nil, errors.NewInternalError(db.Error)
+	}
+	log.Info(ctx, map[string]interface{}{
+		"category_id": category.ID,
+	}, "category created successfully")
+	return category, nil
+}
+
+// LoadAllRelationshipsOfCategory loads all the relationships of a category. This is required for workitemtype filtering.
+func (m *GormRepository) LoadAllRelationshipsOfCategory(ctx context.Context, categoryID uuid.UUID) ([]*WorkItemTypeCategoryRelationship, error) {
+	// Check if category is present
+	_, err := m.LoadCategoryFromDB(ctx, categoryID)
+	if err != nil {
+		return nil, errs.Wrap(err, fmt.Sprintf("failed to load category with id %s", categoryID))
+	}
+
+	relationship := []*WorkItemTypeCategoryRelationship{}
```

Comment:
@kwk I am sorry. Its `Work Item Link Category` (https://github.com/almighty/almighty-core/blob/master/workitem/link/category.go#L16). 

I don't understand your concern here -> 

> Why are there only WorkItemType Relationships and no other? Are others planned?


Could you please ellaborate?